File: codes/cpp/Code_3_01_2.cpp
bool isprime(long long N) {
	// N ‚ğ 2 ˆÈã‚Ì®”‚Æ‚µAN ‚ª‘f”‚Å‚ ‚ê‚Î trueA‘f”‚Å‚È‚¯‚ê‚Î false ‚ğ•Ô‚·
	for (long long i = 2; i * i <= N; i++) {
		if (N % i == 0) return false;
	}
	return true;
}

File: codes/python/Code_3_01_2.py
# 2 以上の整数 N に対し、N が素数であれば true、素数でなければ false を返す関数
def isprime(N):
	LIMIT = int(N ** 0.5)
	for i in range(2, LIMIT + 1):
		if N % i == 0:
			return False
	return True

N = int(input())
if isprime(N):
	print("prime")
else:
	print("not prime")
